presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -399,7 +400,7 @@ public PlanOptimizers(
                                 new TransformCorrelatedSingleRowSubqueryToProject(),
                                 new RemoveAggregationInSemiJoin())),
                 new CheckSubqueryNodesAreRewritten(),
-                predicatePushDown,
+                predicatePushDownWithoutTableProperties,
```

Comment:
Since it's only use in one place, let's inline it. (Side question... why is it only in one place?)

Reply:
Sorry, didn't mean why not insert `PredicatePushdown` in more places, but why only change this one (from the ones that currently exist) to use TableProperties? BTW, there's another one that runs before `PushPredicateIntoTableScan` in line 463.

I believe the answer is that `getTableProperties` can be expensive if no predicates have been pushed down yet and we don't have any other mechanism to protect against that, so we need to insert usages carefully. (Let's add a comment)